cumulusci/core/dependencies/dependencies.py

```diff
@@ -0,0 +1,978 @@
+import abc
+import itertools
+import logging
+from enum import Enum
+from typing import Iterable, List, Optional, Tuple
+
+import pydantic
+from github3.exceptions import NotFoundError
+from github3.repos.repo import Repository
+from pydantic.networks import AnyUrl
+
+from cumulusci.core.config import OrgConfig
+from cumulusci.core.config.project_config import BaseProjectConfig
+from cumulusci.core.dependencies.github import (
+    get_package_data,
+    get_remote_project_config,
+    get_repo,
+)
+from cumulusci.core.dependencies.utils import TaskContext
+from cumulusci.core.exceptions import (
+    CumulusCIException,
+    DependencyParseError,
+    DependencyResolutionError,
+)
+from cumulusci.core.github import (
+    find_latest_release,
+    find_repo_2gp_context,
+    find_repo_feature_prefix,
+    get_version_id_from_commit,
+)
+from cumulusci.salesforce_api.metadata import ApiDeploy
+from cumulusci.salesforce_api.package_install import (
+    DEFAULT_PACKAGE_RETRY_OPTIONS,
+    PackageInstallOptions,
+    install_package_by_namespace_version,
+    install_package_by_version_id,
+)
+from cumulusci.salesforce_api.package_zip import MetadataPackageZipBuilder
+from cumulusci.utils import download_extract_github_from_repo, download_extract_zip
+from cumulusci.utils.git import (
+    construct_release_branch_name,
+    get_feature_branch_name,
+    get_release_identifier,
+    is_release_branch_or_child,
+    split_repo_url,
+)
+from cumulusci.utils.yaml.model_parser import CCIModel
+
+logger = logging.getLogger(__name__)
+
+
+class HashableBaseModel(CCIModel):
+    """Base Pydantic model class that has a functional `hash()` method.
+    Requires that model can be converted to JSON."""
+
+    # See https://github.com/samuelcolvin/pydantic/issues/1303
+    def __hash__(self):
+        return hash((type(self),) + tuple(self.json()))
+
+
+class DependencyResolutionStrategy(str, Enum):
+    """Enum that defines a strategy for resolving a dynamic dependency into a static dependency."""
+
+    STRATEGY_STATIC_TAG_REFERENCE = "tag"
+    STRATEGY_COMMIT_STATUS_EXACT_BRANCH = "commit_status_exact_branch"
+    STRATEGY_COMMIT_STATUS_RELEASE_BRANCH = "commit_status_release_branch"
+    STRATEGY_COMMIT_STATUS_PREVIOUS_RELEASE_BRANCH = (
+        "commit_status_previous_release_branch"
+    )
+    STRATEGY_BETA_RELEASE_TAG = "latest_beta"
+    STRATEGY_RELEASE_TAG = "latest_release"
+    STRATEGY_UNMANAGED_HEAD = "unmanaged"
```

Comment:
seems redundant to prefix these with STRATEGY_; they're already identified as strategies by being accessed as members of DependencyResolutionStrategy

Reply:
Agreed. Changed.